python/cudf/cudf/core/index.py

```diff
@@ -334,13 +334,16 @@ def __contains__(self, item):
             item, tuple(np.sctypes["int"] + np.sctypes["float"] + [int, float])
         ):
             return False
+        elif isinstance(item, bool):
+            return False
         try:
-            item = pd.core.dtypes.common.ensure_python_int(item)
-        except TypeError:
+            int_item = int(item)
+            assert int_item == item
+        except (TypeError, AssertionError):
             return False
```

Comment:
Please don't use exceptions for control-flow (not least `python -O` will remove the `assert` line):

I think, once the boolean case has been handled, the whole body of the function can be:
```
try:
    int_item = int(item)
    return int_item == item and int_item in self._range
except (ValueError, OverflowError):
    return False
```

The `ValueError` handles `nan`, the `OverflowError` handles `inf`.

So the body could be:
```
if isinstance(item, bool) or not isinstance(item, tuple(np.sctypes["int"] + np.sctypes["float"] + [int, float])
    return False
try:
    int_item = int(item)
    return int_item == item and int_item in self._range
except (ValueError, OverflowError):
    return False
```